app/src/main/java/com/remindme/app/screens/timeline/TimelineScreen.kt

```diff
@@ -16,114 +16,176 @@
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.navigation.NavController
 import androidx.navigation.compose.rememberNavController
 import com.remindme.app.ui.theme.RemindMeTheme
+import com.remindme.app.ui.theme.Teal700
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 import java.util.*
 
+@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
 @Composable
 fun ShowTimelineScreen(navController: NavController) {
   RemindMeTheme {
-    Surface {
-      Column(
-        modifier = Modifier
-          .verticalScroll(rememberScrollState())
-          .fillMaxSize()
-      ) {
-
-        Spacer(modifier = Modifier.size(20.dp))
-
-        TimelineListGroupView(
-          list = listOf(
-            Reminder(
-              "Dark Soy Sauce",
-              "Short notes to be added here...",
-              Date(),
-              Date(),
-              Reminder.Settings(
-                emptyList(), Reminder.Notify.Default
-              )
-            ),
-            Reminder(
-              "Calpol Syrup",
-              "Short notes to be added here...",
-              Date(),
-              Date(),
-              Reminder.Settings(
-                emptyList(), Reminder.Notify.Default
-              )
+
+    val scaffoldState = rememberScaffoldState(rememberDrawerState(DrawerValue.Closed))
+    val scope = rememberCoroutineScope()
+
+    Scaffold(
+      scaffoldState = scaffoldState,
+      topBar = { AppTopBar(scaffoldState, scope) },
+      floatingActionButtonPosition = FabPosition.End,
+      floatingActionButton = {
+        FloatingActionButton(onClick = {
+          // TODO open Add new bottom sheet
+        }) {
+          Icon(Icons.Filled.Add, null)
+        }
+      },
+      drawerContent = { LeftDrawerContent() },
+      content = { Content() }
+    )
+  }
+}
+
+@Composable
+fun AppTopBar(scaffoldState: ScaffoldState, scope: CoroutineScope) {
+  TopAppBar(
+    elevation = 5.dp,
+    title = { Text("Timeline") },
+    backgroundColor = Teal700,
+    navigationIcon = {
+      IconButton(onClick = {
+        scope.launch {
+          scaffoldState.drawerState.open()
+        }
+      }) {
+        Icon(Icons.Filled.Menu, null)
+      }
+    },
+    actions = {
+      IconButton(onClick = {
+        // TODO open search screen
+      }) {
+        Icon(Icons.Filled.Search, null)
+      }
+    }
+  )
+}
+
+@Composable
+fun LeftDrawerContent() {
+  // TODO left drawer content UI
+}
+
+@Composable
+fun Content() {
+  Surface {
+    Column(
+      modifier = Modifier
+        .verticalScroll(rememberScrollState())
+        .fillMaxSize()
+    ) {
+
+      Spacer(modifier = Modifier.size(20.dp))
+
+      TimelineListGroupView(
+        list = listOf(
+          Reminder(
+            "Dark Soy Sauce",
```

Comment:
just a thought: if we can use name of parameter, it will improve code readability.